### PR TITLE
release-notes.txt: add 2021.07 release notes

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

Here are the preliminary release notes. Please rearrange/fix/proofread as deemed necessary!

### CODE REVIEW

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,565 @@
...
+
+About this release
+==================
+
+The 2021.07 release includes:
+
+- Many improvements to ztimer usage
+- Added nucleo-wl55jc and Seeeduino XIAO boards
+- stm32f1 kconfig support
+- overflow fix for malloc thread safety
```
Comment (maribu) [inline release-notes.txt:26]: Maybe we should throw in the buzz world BadAlloc (see https://msrc-blog.microsoft.com/2021/04/29/badalloc-memory-allocation-vulnerabilities-could-affect-wide-range-of-iot-and-ot-devices-in-industrial-medical-and-enterprise-networks/)

And for ESP, which is not using the wrapper for thread safe allocations, the fix is done separately. Maybe we could just have something along the lines of: "Fixed multiplication overflow in calloc, a.k.a. BadAlloc, for all supported platforms". The implementation details come later on in the list of merged PRs, so IMO we could just go with a superficial one-liner here.

**Review by maribu — COMMENTED** [review]
see inline, otherwise all looks good to me

File: release-notes.txt (hunk 1 above)
Comment (MrKevinWeiss) [reply release-notes.txt]: Updated and directly amended

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,567 @@
...
+==================
+
+The 2021.07 release includes:
+
+- Many improvements to ztimer usage
+- Added nucleo-wl55jc and Seeeduino XIAO boards
+- stm32f1 kconfig support
+- BadAlloc fix for malloc thread safety
+- Fixed multiplication overflow in calloc, a.k.a. BadAlloc, for all supported
+platforms
```
Comment (maribu) [inline release-notes.txt:28]:
```suggestion
- Fixed multiplication overflow in calloc, a.k.a. BadAlloc, for all supported
  platforms
```

**Review by maribu — APPROVED** [review]
One formatation nitpick (please squash right away). Otherwise, this looks good to me. Thanks!

File: release-notes.txt (hunk 2 above)
Comment (MrKevinWeiss) [reply release-notes.txt:28]: done

Comment (maribu) [reply release-notes.txt:28]: Hmm, looks like the previous state got push again...

**Comment by MrKevinWeiss** [comment]
cough cough cough...

File: release-notes.txt [hunk 3]
```diff
@@ -1,3 +1,567 @@
...
+==================
+
+The 2021.07 release includes:
+
+- Many improvements to ztimer usage
+- Added nucleo-wl55jc and Seeeduino XIAO boards
+- stm32f1 kconfig support
+- BadAlloc fix for malloc thread safety
+- Fixed multiplication overflow in calloc, a.k.a. BadAlloc, for all supported
+  platforms
```
Comment (leandrolanzieri) [inline release-notes.txt:28]: @maribu are you ok with these two points like so?

Comment (leandrolanzieri) [reply release-notes.txt]: Ok, makes sense now

**Review by leandrolanzieri — APPROVED** [review]
Looks good, ACK!

**Review by jia200x — APPROVED** [review]
Here's my ACK too

**Pull request merged**